warehouse/admin/views/prohibited_email_domains.py

```diff
@@ -0,0 +1,131 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from paginate_sqlalchemy import SqlalchemyOrmPage as SQLAlchemyORMPage
+from pyramid.httpexceptions import HTTPBadRequest, HTTPSeeOther
+from pyramid.view import view_config
+from sqlalchemy import func
+from tldextract import extract
+
+from warehouse.accounts.models import ProhibitedEmailDomain
+from warehouse.authnz import Permissions
+from warehouse.utils.paginate import paginate_url_factory
+
+
+@view_config(
+    route_name="admin.prohibited_email_domains.list",
+    renderer="admin/prohibited_email_domains/list.html",
+    permission=Permissions.AdminProhibitedEmailDomainsRead,
+    request_method="GET",
+    uses_session=True,
+)
+def prohibited_email_domains(request):
+    q = request.params.get("q")
+
+    try:
+        page_num = int(request.params.get("page", 1))
+    except ValueError:
+        raise HTTPBadRequest("'page' must be an integer.") from None
+
+    prohibited_email_domains_query = request.db.query(ProhibitedEmailDomain).order_by(
+        ProhibitedEmailDomain.created.desc()
+    )
+
+    if q:
+        prohibited_email_domains_query = prohibited_email_domains_query.filter(
+            ProhibitedEmailDomain.domain.ilike(q)
+        )
+
+    prohibited_email_domains = SQLAlchemyORMPage(
+        prohibited_email_domains_query,
+        page=page_num,
+        items_per_page=25,
+        url_maker=paginate_url_factory(request),
+    )
+
+    return {"prohibited_email_domains": prohibited_email_domains, "query": q}
+
+
+@view_config(
+    route_name="admin.prohibited_email_domains.add",
+    permission=Permissions.AdminProhibitedEmailDomainsWrite,
+    request_method="POST",
+    uses_session=True,
+    require_methods=False,
+)
+def add_prohibited_email_domain(request):
```

Comment:
Do we care if the domain has a protocol or not? Looks like this will accept either:

<img width="1243" alt="image" src="https://github.com/user-attachments/assets/d97bb17b-dff0-4c04-8aec-b82640378dda">

Reply:
Good call-out! We should remove some of the stuff we don't care about after it's been posted, so we only store parts of the input we care about and would compare against during validation.

Reply:
Updated in e60459158541b6538785fcc597a4f0106a493b45